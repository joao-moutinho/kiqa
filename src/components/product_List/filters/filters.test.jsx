/* eslint-disable testing-library/prefer-presence-queries */
/* eslint-disable jest/valid-expect */
/* eslint-disable testing-library/no-container */
/* eslint-disable no-undef */
import { fireEvent, render, renderHook, screen, waitForElement } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { QueryClient, QueryClientProvider, useQuery } from "react-query";
import { BrowserRouter } from "react-router-dom";
import { Filters } from ".";

describe("Filters", () => {
  const queryClient = new QueryClient();

  test("render", async () => {
   /*  jest.mock("react-query", () => ({
      useQuery: () => ({ isLoading: false, error: {}, data: [] }),
    })); */

    const fn = jest.fn();

    const { container } = render(
      <QueryClientProvider client={queryClient}>
        <BrowserRouter>
          <Filters  searchParamsChange={fn}/>
        </BrowserRouter>
      </QueryClientProvider>
    );
    expect(container).toMatchSnapshot();
  });

  test("should change value when move mouse", async () => {
    const fn = jest.fn();
    const { container } = render(
      <QueryClientProvider client={queryClient}>
        <BrowserRouter>
          <Filters  searchParamsChange={fn}/>
        </BrowserRouter>
      </QueryClientProvider>
    );


    const slider =  screen.getAllByRole("slider");
    // eslint-disable-next-line no-undef

    fireEvent.mouseDown(slider[0], {clientX: 10});
    fireEvent.mouseMove(slider[0], {clienteX: 30});
    fireEvent.mouseUp(slider[0]);

    fireEvent.mouseDown(slider[1], {clientX: 60});
    fireEvent.mouseMove(slider[1], {clienteX: 30});
    fireEvent.mouseUp(slider[1]);

    expect(container).toMatchSnapshot();
 
  })

  test('should show categories when pressed chevron', () => {
    const fn = jest.fn();
    const { container } = render(
      <QueryClientProvider client={queryClient}>
        <BrowserRouter>
          <Filters  searchParamsChange={fn}/>
        </BrowserRouter>
      </QueryClientProvider>
    );

    fireEvent.click(screen.getByTestId("chevron-categories"));
    expect(screen.getByTestId("showCategories")).toMatchSnapshot();
    fireEvent.click(screen.getByTestId("chevron-categories"));
    expect(container).toMatchSnapshot();
  })

});
