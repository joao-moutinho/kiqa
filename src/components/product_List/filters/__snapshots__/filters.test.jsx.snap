// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filters render 1`] = `
<div>
  <div
    class="filters"
  >
    <div
      class="filters-price"
    >
      <div
        class="filters-price-text"
      >
        Price
      </div>
      <div
        class="filters-price-range"
      >
        <div
          class="css-15g0dol-Base e1n4b2jv0"
        >
          <div
            class="css-i35pig-Slider e1p4hzk20"
          >
            <div
              class="rheostat rheostat-horizontal "
              style="position: relative;"
            >
              <div
                class="rheostat-background"
              />
              <button
                aria-disabled="false"
                aria-label="slider-button"
                aria-valuemax="77"
                aria-valuemin="1.99"
                aria-valuenow="1.99"
                class="rheostat-handle"
                data-handle-key="0"
                role="slider"
                style="left: 0%; position: absolute;"
                tabindex="0"
              />
              <button
                aria-disabled="false"
                aria-label="slider-button"
                aria-valuemax="77"
                aria-valuemin="1.99"
                aria-valuenow="77"
                class="rheostat-handle"
                data-handle-key="1"
                role="slider"
                style="left: 100%; position: absolute;"
                tabindex="0"
              />
              <div
                class="rheostat-progress"
                style="left: 0%; width: 100%;"
              />
            </div>
            <div
              class="css-dqmsxc-rangeLabelsContainer"
            >
              <div
                class="css-nb97e2-Label-left e1atnqi30"
              >
                1 €
              </div>
              <div
                class="css-qjht2a-Label-right e1atnqi30"
              >
                80 €
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="filters-type"
    >
      <div
        class="filters-type-title"
      >
        <svg
          class="filters-type-title-iconclose"
          data-testid="chevron-categories"
          fill="currentColor"
          height="30"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 24 24"
          width="30"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M16.293 9.293 12 13.586 7.707 9.293l-1.414 1.414L12 16.414l5.707-5.707z"
          />
        </svg>
        <p>
          Categories
        </p>
      </div>
    </div>
    <div
      class="filters-subCategory"
    >
      <div
        class="filters-type-title"
      >
        <svg
          class="filters-type-title-iconopen"
          fill="currentColor"
          height="30"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 24 24"
          width="30"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M16.293 9.293 12 13.586 7.707 9.293l-1.414 1.414L12 16.414l5.707-5.707z"
          />
        </svg>
        <p>
          Sub-Category
        </p>
      </div>
      <div
        class="filters-category-checkbox"
      >
        <div
          class="filters-type-checkbox-input"
        >
          <input
            disabled=""
            name="brand"
            type="checkbox"
          />
          <label
            for="nail_polish"
          >
            nail_polish
          </label>
        </div>
        <div
          class="filters-type-checkbox-input"
        >
          <input
            disabled=""
            name="brand"
            type="checkbox"
          />
          <label
            for="blush"
          >
            blush
          </label>
        </div>
        <div
          class="filters-type-checkbox-input"
        >
          <input
            disabled=""
            name="brand"
            type="checkbox"
          />
          <label
            for="lipstick"
          >
            lipstick
          </label>
        </div>
        <div
          class="filters-type-checkbox-input"
        >
          <input
            disabled=""
            name="brand"
            type="checkbox"
          />
          <label
            for="mascara"
          >
            mascara
          </label>
        </div>
        <div
          class="filters-type-checkbox-input"
        >
          <input
            disabled=""
            name="brand"
            type="checkbox"
          />
          <label
            for="lip_liner"
          >
            lip_liner
          </label>
        </div>
        <div
          class="filters-type-checkbox-input"
        >
          <input
            disabled=""
            name="brand"
            type="checkbox"
          />
          <label
            for="foundation"
          >
            foundation
          </label>
        </div>
        <div
          class="filters-type-checkbox-input"
        >
          <input
            disabled=""
            name="brand"
            type="checkbox"
          />
          <label
            for="eyeshadow"
          >
            eyeshadow
          </label>
        </div>
        <div
          class="filters-type-checkbox-input"
        >
          <input
            disabled=""
            name="brand"
            type="checkbox"
          />
          <label
            for="eyeliner"
          >
            eyeliner
          </label>
        </div>
        <div
          class="filters-type-checkbox-input"
        >
          <input
            disabled=""
            name="brand"
            type="checkbox"
          />
          <label
            for="eyebrow"
          >
            eyebrow
          </label>
        </div>
        <div
          class="filters-type-checkbox-input"
        >
          <input
            disabled=""
            name="brand"
            type="checkbox"
          />
          <label
            for="bronzer"
          >
            bronzer
          </label>
        </div>
      </div>
    </div>
    <div
      class="filters-category"
    >
      <div
        class="filters-type-title"
      >
        <svg
          class="filters-type-title-iconclose"
          fill="currentColor"
          height="30"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 24 24"
          width="30"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M16.293 9.293 12 13.586 7.707 9.293l-1.414 1.414L12 16.414l5.707-5.707z"
          />
        </svg>
        <p>
          Brands
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`Filters should change value when move mouse 1`] = `
<div>
  <div
    class="filters"
  >
    <div
      class="filters-price"
    >
      <div
        class="filters-price-text"
      >
        Price
      </div>
      <div
        class="filters-price-range"
      >
        <div
          class="css-15g0dol-Base e1n4b2jv0"
        >
          <div
            class="css-i35pig-Slider e1p4hzk20"
          >
            <div
              class="rheostat rheostat-horizontal "
              style="position: relative;"
            >
              <div
                class="rheostat-background"
              />
              <button
                aria-disabled="false"
                aria-label="slider-button"
                aria-valuemax="77"
                aria-valuemin="1.99"
                aria-valuenow="1.99"
                class="rheostat-handle"
                data-handle-key="0"
                role="slider"
                style="left: 0%; position: absolute;"
                tabindex="0"
              />
              <button
                aria-disabled="false"
                aria-label="slider-button"
                aria-valuemax="77"
                aria-valuemin="1.99"
                aria-valuenow="77"
                class="rheostat-handle"
                data-handle-key="1"
                role="slider"
                style="left: 100%; position: absolute;"
                tabindex="0"
              />
              <div
                class="rheostat-progress"
                style="left: 0%; width: 100%;"
              />
            </div>
            <div
              class="css-dqmsxc-rangeLabelsContainer"
            >
              <div
                class="css-nb97e2-Label-left e1atnqi30"
              >
                NaN €
              </div>
              <div
                class="css-qjht2a-Label-right e1atnqi30"
              >
                77 €
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="filters-type"
    >
      <div
        class="filters-type-title"
      >
        <svg
          class="filters-type-title-iconclose"
          data-testid="chevron-categories"
          fill="currentColor"
          height="30"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 24 24"
          width="30"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M16.293 9.293 12 13.586 7.707 9.293l-1.414 1.414L12 16.414l5.707-5.707z"
          />
        </svg>
        <p>
          Categories
        </p>
      </div>
    </div>
    <div
      class="filters-subCategory"
    >
      <div
        class="filters-type-title"
      >
        <svg
          class="filters-type-title-iconopen"
          fill="currentColor"
          height="30"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 24 24"
          width="30"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M16.293 9.293 12 13.586 7.707 9.293l-1.414 1.414L12 16.414l5.707-5.707z"
          />
        </svg>
        <p>
          Sub-Category
        </p>
      </div>
      <div
        class="filters-category-checkbox"
      >
        <div
          class="filters-type-checkbox-input"
        >
          <input
            disabled=""
            name="brand"
            type="checkbox"
          />
          <label
            for="nail_polish"
          >
            nail_polish
          </label>
        </div>
        <div
          class="filters-type-checkbox-input"
        >
          <input
            disabled=""
            name="brand"
            type="checkbox"
          />
          <label
            for="blush"
          >
            blush
          </label>
        </div>
        <div
          class="filters-type-checkbox-input"
        >
          <input
            disabled=""
            name="brand"
            type="checkbox"
          />
          <label
            for="lipstick"
          >
            lipstick
          </label>
        </div>
        <div
          class="filters-type-checkbox-input"
        >
          <input
            disabled=""
            name="brand"
            type="checkbox"
          />
          <label
            for="mascara"
          >
            mascara
          </label>
        </div>
        <div
          class="filters-type-checkbox-input"
        >
          <input
            disabled=""
            name="brand"
            type="checkbox"
          />
          <label
            for="lip_liner"
          >
            lip_liner
          </label>
        </div>
        <div
          class="filters-type-checkbox-input"
        >
          <input
            disabled=""
            name="brand"
            type="checkbox"
          />
          <label
            for="foundation"
          >
            foundation
          </label>
        </div>
        <div
          class="filters-type-checkbox-input"
        >
          <input
            disabled=""
            name="brand"
            type="checkbox"
          />
          <label
            for="eyeshadow"
          >
            eyeshadow
          </label>
        </div>
        <div
          class="filters-type-checkbox-input"
        >
          <input
            disabled=""
            name="brand"
            type="checkbox"
          />
          <label
            for="eyeliner"
          >
            eyeliner
          </label>
        </div>
        <div
          class="filters-type-checkbox-input"
        >
          <input
            disabled=""
            name="brand"
            type="checkbox"
          />
          <label
            for="eyebrow"
          >
            eyebrow
          </label>
        </div>
        <div
          class="filters-type-checkbox-input"
        >
          <input
            disabled=""
            name="brand"
            type="checkbox"
          />
          <label
            for="bronzer"
          >
            bronzer
          </label>
        </div>
      </div>
    </div>
    <div
      class="filters-category"
    >
      <div
        class="filters-type-title"
      >
        <svg
          class="filters-type-title-iconclose"
          fill="currentColor"
          height="30"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 24 24"
          width="30"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M16.293 9.293 12 13.586 7.707 9.293l-1.414 1.414L12 16.414l5.707-5.707z"
          />
        </svg>
        <p>
          Brands
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`Filters should show categories when pressed chevron 1`] = `
<div
  class="filters-type-checkbox"
  data-testid="showCategories"
>
  <div
    class="filters-type-checkbox-input"
  >
    <input
      name="lips"
      type="checkbox"
    />
    <label
      for="lips"
    >
      lips
    </label>
  </div>
  <div
    class="filters-type-checkbox-input"
  >
    <input
      name="eyes"
      type="checkbox"
    />
    <label
      for="eyes"
    >
      eyes
    </label>
  </div>
  <div
    class="filters-type-checkbox-input"
  >
    <input
      name="face"
      type="checkbox"
    />
    <label
      for="face"
    >
      face
    </label>
  </div>
  <div
    class="filters-type-checkbox-input"
  >
    <input
      name="nails"
      type="checkbox"
    />
    <label
      for="nails"
    >
      nails
    </label>
  </div>
</div>
`;

exports[`Filters should show categories when pressed chevron 2`] = `
<div>
  <div
    class="filters"
  >
    <div
      class="filters-price"
    >
      <div
        class="filters-price-text"
      >
        Price
      </div>
      <div
        class="filters-price-range"
      >
        <div
          class="css-15g0dol-Base e1n4b2jv0"
        >
          <div
            class="css-i35pig-Slider e1p4hzk20"
          >
            <div
              class="rheostat rheostat-horizontal "
              style="position: relative;"
            >
              <div
                class="rheostat-background"
              />
              <button
                aria-disabled="false"
                aria-label="slider-button"
                aria-valuemax="77"
                aria-valuemin="1.99"
                aria-valuenow="1.99"
                class="rheostat-handle"
                data-handle-key="0"
                role="slider"
                style="left: 0%; position: absolute;"
                tabindex="0"
              />
              <button
                aria-disabled="false"
                aria-label="slider-button"
                aria-valuemax="77"
                aria-valuemin="1.99"
                aria-valuenow="77"
                class="rheostat-handle"
                data-handle-key="1"
                role="slider"
                style="left: 100%; position: absolute;"
                tabindex="0"
              />
              <div
                class="rheostat-progress"
                style="left: 0%; width: 100%;"
              />
            </div>
            <div
              class="css-dqmsxc-rangeLabelsContainer"
            >
              <div
                class="css-nb97e2-Label-left e1atnqi30"
              >
                1 €
              </div>
              <div
                class="css-qjht2a-Label-right e1atnqi30"
              >
                80 €
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="filters-type"
    >
      <div
        class="filters-type-title"
      >
        <svg
          class="filters-type-title-iconclose"
          data-testid="chevron-categories"
          fill="currentColor"
          height="30"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 24 24"
          width="30"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M16.293 9.293 12 13.586 7.707 9.293l-1.414 1.414L12 16.414l5.707-5.707z"
          />
        </svg>
        <p>
          Categories
        </p>
      </div>
    </div>
    <div
      class="filters-subCategory"
    >
      <div
        class="filters-type-title"
      >
        <svg
          class="filters-type-title-iconopen"
          fill="currentColor"
          height="30"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 24 24"
          width="30"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M16.293 9.293 12 13.586 7.707 9.293l-1.414 1.414L12 16.414l5.707-5.707z"
          />
        </svg>
        <p>
          Sub-Category
        </p>
      </div>
      <div
        class="filters-category-checkbox"
      >
        <div
          class="filters-type-checkbox-input"
        >
          <input
            disabled=""
            name="brand"
            type="checkbox"
          />
          <label
            for="nail_polish"
          >
            nail_polish
          </label>
        </div>
        <div
          class="filters-type-checkbox-input"
        >
          <input
            disabled=""
            name="brand"
            type="checkbox"
          />
          <label
            for="blush"
          >
            blush
          </label>
        </div>
        <div
          class="filters-type-checkbox-input"
        >
          <input
            disabled=""
            name="brand"
            type="checkbox"
          />
          <label
            for="lipstick"
          >
            lipstick
          </label>
        </div>
        <div
          class="filters-type-checkbox-input"
        >
          <input
            disabled=""
            name="brand"
            type="checkbox"
          />
          <label
            for="mascara"
          >
            mascara
          </label>
        </div>
        <div
          class="filters-type-checkbox-input"
        >
          <input
            disabled=""
            name="brand"
            type="checkbox"
          />
          <label
            for="lip_liner"
          >
            lip_liner
          </label>
        </div>
        <div
          class="filters-type-checkbox-input"
        >
          <input
            disabled=""
            name="brand"
            type="checkbox"
          />
          <label
            for="foundation"
          >
            foundation
          </label>
        </div>
        <div
          class="filters-type-checkbox-input"
        >
          <input
            disabled=""
            name="brand"
            type="checkbox"
          />
          <label
            for="eyeshadow"
          >
            eyeshadow
          </label>
        </div>
        <div
          class="filters-type-checkbox-input"
        >
          <input
            disabled=""
            name="brand"
            type="checkbox"
          />
          <label
            for="eyeliner"
          >
            eyeliner
          </label>
        </div>
        <div
          class="filters-type-checkbox-input"
        >
          <input
            disabled=""
            name="brand"
            type="checkbox"
          />
          <label
            for="eyebrow"
          >
            eyebrow
          </label>
        </div>
        <div
          class="filters-type-checkbox-input"
        >
          <input
            disabled=""
            name="brand"
            type="checkbox"
          />
          <label
            for="bronzer"
          >
            bronzer
          </label>
        </div>
      </div>
    </div>
    <div
      class="filters-category"
    >
      <div
        class="filters-type-title"
      >
        <svg
          class="filters-type-title-iconclose"
          fill="currentColor"
          height="30"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 24 24"
          width="30"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M16.293 9.293 12 13.586 7.707 9.293l-1.414 1.414L12 16.414l5.707-5.707z"
          />
        </svg>
        <p>
          Brands
        </p>
      </div>
    </div>
  </div>
</div>
`;
